crates/rpc/rpc-eth-api/src/helpers/pending_block.rs

```diff
@@ -163,7 +163,25 @@ pub trait LoadPendingBlock: EthApiTypes {
             };
 
             let now = Instant::now();
-            *lock = Some(PendingBlock::new(pending_block.clone(), now + Duration::from_secs(1)));
+
+            // Fetch receipts for the pending block
+            let receipts = match self.provider().receipts_by_block_id(pending_block.number.into()) {
+                Ok(Some(receipts)) => receipts,
+                Ok(None) => {
+                    debug!(target: "rpc", "No receipts found for pending block");
+                    Vec::new() 
+                }
+                Err(err) => {
+                    debug!(target: "rpc", "Failed to fetch receipts for pending block: {:?}", err);
+                    return Err(Self::Error::from_eth_err(EthApiError::Internal(err.into())))
+                }
+            };
```

Comment:
this doesn't really work because we're trying to fetch a block by number that doesn't exist.

what we need to do is change how this behaves:

https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc-eth-api/src/helpers/pending_block.rs#L219-L228

this should also return the receipts